src/modules/mavlink/mavlink_messages.cpp

```diff
@@ -3954,186 +3966,607 @@ class MavlinkStreamHighLatency : public MavlinkStream
+			}
+
+			if (((status.onboard_control_sensors_enabled & MAV_SYS_STATUS_SENSOR_3D_GYRO)
+			     && !(status.onboard_control_sensors_health & MAV_SYS_STATUS_SENSOR_3D_GYRO)) ||
+			    ((status.onboard_control_sensors_enabled & MAV_SYS_STATUS_SENSOR_3D_GYRO2) &&
+			     !(status.onboard_control_sensors_health & MAV_SYS_STATUS_SENSOR_3D_GYRO2))) {
+				msg->failure_flags |= HL_FAILURE_FLAG_3D_GYRO;
+			}
+
+			if (((status.onboard_control_sensors_enabled & MAV_SYS_STATUS_SENSOR_3D_MAG)
+			     && !(status.onboard_control_sensors_health & MAV_SYS_STATUS_SENSOR_3D_MAG)) ||
+			    ((status.onboard_control_sensors_enabled & MAV_SYS_STATUS_SENSOR_3D_MAG2) &&
+			     !(status.onboard_control_sensors_health & MAV_SYS_STATUS_SENSOR_3D_MAG2))) {
+				msg->failure_flags |= HL_FAILURE_FLAG_3D_MAG;
+			}
+
+			if ((status.onboard_control_sensors_enabled & MAV_SYS_STATUS_TERRAIN)
+			    && !(status.onboard_control_sensors_health & MAV_SYS_STATUS_TERRAIN)) {
+				msg->failure_flags |= HL_FAILURE_FLAG_TERRAIN;
+			}
+
+			if (status.rc_signal_lost) {
+				msg->failure_flags |= HL_FAILURE_FLAG_RC_RECEIVER;
+			}
+
+			if (status.engine_failure) {
+				msg->failure_flags |= HL_FAILURE_FLAG_ENGINE;
+			}
+
+			if (status.mission_failure) {
+				msg->failure_flags |= HL_FAILURE_FLAG_MISSION;
+			}
+
+			// flight mode
+			union px4_custom_mode custom_mode;
+			uint8_t mavlink_base_mode;
+			get_mavlink_custom_mode(&status, &mavlink_base_mode, &custom_mode);
+			msg->custom_mode = custom_mode.custom_mode_hl;
+		}
+
+		return updated;
+	}
+
+	bool write_vehicle_status_flags(mavlink_high_latency2_t *msg)
+	{
+		struct vehicle_status_flags_s status_flags = {};
+
+		const bool updated = _status_flags_sub->update(&_status_flags_time, &status_flags);
+
+		if (_status_flags_time > 0) {
+			if (!status_flags.condition_global_position_valid) { //TODO check if there is a better way to get only GPS failure
+				msg->failure_flags |= HL_FAILURE_FLAG_GPS;
+			}
+
+			if (status_flags.offboard_control_signal_lost && status_flags.offboard_control_signal_found_once) {
+				msg->failure_flags |= HL_FAILURE_FLAG_OFFBOARD_LINK;
+			}
+		}
+
+		return updated;
+	}
+
+	bool write_wind_estimate(mavlink_high_latency2_t *msg)
+	{
+		struct wind_estimate_s wind = {};
+
+		const bool updated = _wind_sub->update(&_wind_time, &wind);
+
+		if (_wind_time > 0) {
+			msg->wind_heading = static_cast<uint8_t>(
+						    math::degrees(_wrap_2pi(atan2f(wind.windspeed_east, wind.windspeed_north))) * 0.5f);
 		}
 
 		return updated;
 	}
+
+	void update_data()
+	{
+		const hrt_abstime t = hrt_absolute_time();
+
+		if (t > _last_update_time) {
+			// first order low pass filter for the update rate
+			_update_rate_filtered = 0.97f * _update_rate_filtered + 0.03f / ((t - _last_update_time) * 1e-6f);
+			_last_update_time = t;
+		}
+
+		update_airspeed(_update_rate_filtered);
+
+		update_tecs_status(_update_rate_filtered);
+
+		update_battery_status(_update_rate_filtered);
```

Comment:
Since you're passing a class member `_update_rate_filtered`, you can avoid adding an argument and use `_update_rate_filtered` directly in `update_battery_status`

Reply:
Done